### PR TITLE
feat: List record with `include` vectors parameter

### LINKED ISSUE #4102: [FEATURE] Fetching vectors with records from Python SDK (`with_vectors=["vector1", "vector2",...`)

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR supports fetching records with vectors from Python SDK. A new argument is provided in `from_argilla` method to provide which vectors must be downloaded with records (None by default). Users can set this argument to `all` to download all vectors found in the records.

I also the relationship in the `Vector` ORM class to load `vector_settings` has been changed to load them by default. Otherwise, a massive `awaitable_attrs` must be applied in order to create the vector dictionary without getting the lovely error `sqlalchemy.exc.MissingGreenlet: greenlet_spawn has not been called;`

Please, @gabrielmbmb take a look into it, and let's discuss if we can optimize this a bit. 

Closes https://github.com/argilla-io/argilla/issues/4102

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/feedback/dataset/local/mixins.py [hunk 1]
```diff
@@ -349,6 +349,7 @@ def from_argilla(
         *,
         workspace: Optional[str] = None,
         id: Optional[Union[UUID, str]] = None,
+        with_vectors: Union[None, INCLUDE_ALL_VECTORS_PARAM, List[str]] = None,
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/local/mixins.py]:
```suggestion
        with_vectors: Union[Literal[INCLUDE_ALL_VECTORS_PARAM], List[str], None] = None,
```

File: src/argilla/client/feedback/dataset/remote/dataset.py [hunk 2]
```diff
@@ -358,6 +390,7 @@ def __init__(
         questions: List["AllowedRemoteQuestionTypes"],
         guidelines: Optional[str] = None,
         allow_extra_metadata: bool = True,
+        with_vectors: Union[INCLUDE_ALL_VECTORS_PARAM, List[str], None] = None,
```
Comment (gabrielmbmb) [inline src/argilla/client/feedback/dataset/remote/dataset.py]:
```suggestion
        with_vectors: Union[Literal[INCLUDE_ALL_VECTORS_PARAM], List[str], None] = None,
```

File: src/argilla/client/sdk/v1/datasets/api.py [hunk 3]
```diff
@@ -186,6 +187,8 @@ def get_records(
     Args:
         client: the authenticated Argilla client to be used to send the request to the API.
         id: the id of the dataset to retrieve the records from.
+        include: the fields to be included in the response.
+             Can either be `responses` or `suggestions`.
```
Comment (gabrielmbmb) [inline src/argilla/client/sdk/v1/datasets/api.py]:
```suggestion
        include: the fields to be included in the response.
             Can either be `responses`, `suggestions`, `vectors:all` or `vectors:vector_name_1,vector_name_2,...`
```

File: src/argilla/server/models/database.py [hunk 4]
```diff
@@ -132,7 +132,7 @@ class Vector(DatabaseModel):
     vector_settings_id: Mapped[UUID] = mapped_column(ForeignKey("vectors_settings.id", ondelete="CASCADE"), index=True)
 
     record: Mapped["Record"] = relationship(back_populates="vectors")
-    vector_settings: Mapped["VectorSettings"] = relationship(back_populates="vectors")
+    vector_settings: Mapped["VectorSettings"] = relationship(back_populates="vectors", lazy="immediate")
```
Comment (gabrielmbmb) [inline src/argilla/server/models/database.py]: I'll try to get rid of this

**Review by gabrielmbmb — APPROVED** [review]
LGTM! I'll try to get rid of the `lazy="inmmediate"`

File: tests/integration/client/sdk/v1/test_records.py [hunk 5]
```diff
@@ -57,6 +57,50 @@ def test_dataset():
 
 @pytest.mark.asyncio
 class TestRecordsSDK:
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    async def test_update_record_with_vectors(self, owner: User, role: UserRole) -> None:
```
Comment (gabrielmbmb) [inline tests/integration/client/sdk/v1/test_records.py]:
```suggestion
    def test_update_record_with_vectors(self, owner: User, role: UserRole) -> None:
```

Comment (frascuchon) [reply tests/integration/client/sdk/v1/test_records.py]: I remove the use of the `UserFactory` to apply your changes.

**Pull request merged**